src/engine/actions/dml/scan.rs

```
@@ -159,27 +204,53 @@ impl DBEngine {
+        let existing_len = if write.replace_existing {
+            0
+        } else {
+            match tokio::fs::metadata(&write.segment_path).await {
+                Ok(metadata) => metadata.len(),
+                Err(error) if error.kind() == IOErrorKind::NotFound => 0,
+                Err(error) => return Err(ExecuteError::wrap(error.to_string())),
+            }
+        };
+
+        let mut file = match tokio::fs::OpenOptions::new()
+            .create(true)
```

Clippy (1.96.0)

[warning] file opened with `create`, but `truncate` behavior not defined

(warning)